scom/stackstate_checks/scom/scom.py

```diff
@@ -82,114 +52,136 @@ def sendAlerts(self, session, scom_ip, domain, username, password, component_id)
                     "time_added:%s" % event.get("timeadded")
                 ]
             })
+    def get_health_state(self,healthIconUrl):
+        if re.search('StatusOkComplete', healthIconUrl, flags=re.IGNORECASE) or re.search('healthy_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Healthy'
+        elif re.search('StatusWarning', healthIconUrl, flags=re.IGNORECASE) or re.search('warning_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Deviating'
+        elif re.search('StatusError', healthIconUrl, flags=re.IGNORECASE) or re.search('critical_ex', healthIconUrl,flags=re.IGNORECASE):
+           return 'Error'
+        elif re.search('StatusUnknownGreen', healthIconUrl, flags=re.IGNORECASE) or re.search('maintenance_ex',healthIconUrl,flags=re.IGNORECASE):
+           return 'Not Monitored'
+        else:
+           return str(healthIconUrl.split("/")[-1])
+
+    def get_type(self,session,component_id,domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        type_data = {
+                                "criteria": "(Id = '"+component_id+"')"
+                                }
+        type_response = (session.post('http://'+ scom_ip +'/OperationsManager/data/scomObjects',auth=HttpNtlmAuth(domain + '\\' + username, password),json=type_data["criteria"])).json()
+        self.requests_counter +=1
+        self.log.debug("Number of requets: "+str(self.requests_counter))
+        component_type = type_response.get("scopeDatas")[0].get("className").lower().replace(" ","-")
+        return str(component_type)
+
+    def get_component_data_and_relations(self,session,component_id, domain, username, password, scom_ip):
+        if self.requests_counter > self.requests_threshold:
+           return
+        response = (session.get('http://' + scom_ip + '/OperationsManager/data/objectInformation/' + component_id, auth=HttpNtlmAuth(domain + '\\' + username, password))).json()
+        self.requests_counter +=1
+        self.log.debug("Number of requets: "+str(self.requests_counter))
+        properties = response.get("monitoringObjectProperties", [])
+        relations = response.get("relatedObjects", [])
+        properties_list = dict()
+        relations_list = []
+        for detail in properties:
+            properties_list.update({str(detail.get("name").encode('utf-8')):str(detail.get("value").encode('utf-8'))})
+        name = str(response["displayName"])
```

Comment:
a few things here that can potentially fail, make sure that all these values are there or use some defaults if possible